lib/postgrex/replication.ex

```diff
@@ -694,140 +439,45 @@ defmodule Postgrex.Replication do
     end
   end
 
-  @doc false
-  def handle_call({:start_replication, opts}, _from, %{repl_opts: nil, copy_opts: nil} = s) do
-    statement = command(:start_replication, opts)
-
-    with {:ok, slot, protocol} <- maybe_create_slot(opts, s.protocol),
-         {:ok, protocol} <- Protocol.handle_replication(statement, protocol),
-         {:ok, protocol} <- Protocol.checkin(protocol) do
-      {:reply, {:ok, slot}, %{s | protocol: protocol, repl_opts: opts}}
-    else
-      {:error, reason, protocol} ->
-        {:reply, {:error, reason}, %{s | protocol: protocol}}
-
-      {:disconnect, reason, protocol} ->
-        reconnect_or_stop(:disconnect, reason, protocol, s)
-    end
-  end
-
-  @doc false
-  def handle_call({:copy_table, opts}, _from, %{repl_opts: nil, copy_opts: nil} = s) do
-    table_name = Keyword.fetch!(opts, :table_name)
-    begin_statement = command(:begin_copy, opts)
-    create_slot_statement = command(:create_slot, opts)
-    copy_statement = command(:copy_table, opts)
-
-    with {:ok, _, protocol} <- Protocol.handle_simple(begin_statement, s.protocol),
-         {:ok, slot, protocol} <- Protocol.handle_simple(create_slot_statement, protocol),
-         {:ok, columns, protocol} <- table_columns(table_name, protocol),
-         {:ok, protocol} <- Protocol.handle_copy_table(copy_statement, protocol),
-         {:ok, protocol} <- Protocol.checkin(protocol) do
-      s = %{s | protocol: protocol, copy_opts: Keyword.put(opts, :columns, columns)}
-      {:reply, {:ok, slot}, s}
-    else
-      {:error, reason, protocol} ->
-        {:reply, {:error, reason}, %{s | protocol: protocol}}
-
-      {:disconnect, reason, protocol} ->
-        reconnect_or_stop(:disconnect, reason, protocol, s)
-    end
-  end
-
-  @doc false
-  def handle_call({name, opts}, _from, %{repl_opts: nil, copy_opts: nil} = s)
-      when name in @public_commands do
-    statement = command(name, opts)
-
-    case Protocol.handle_simple(statement, s.protocol) do
-      {:ok, %Postgrex.Result{} = result, protocol} ->
-        {:reply, {:ok, result}, %{s | protocol: protocol}}
-
-      {:error, reason, protocol} ->
-        {:reply, {:error, reason}, %{s | protocol: protocol}}
-
-      {:disconnect, reason, protocol} ->
-        reconnect_or_stop(:disconnect, reason, protocol, s)
-    end
-  end
-
-  @doc false
-  def handle_call({name, _opts}, _from, s) when name in @public_commands,
-    do: {:reply, {:error, :stream_in_progress}, s}
-
-  @doc false
   def handle_call(msg, from, %{state: {mod, mod_state}} = s) do
-    handle(mod, :handle_call, [msg, from, mod_state], s)
+    handle(mod, :handle_call, [msg, from, mod_state], from, s)
```

Comment:
`handle_call` is listed as an optional callback and this is using `handle` rather than `maybe_handle`. It makes sense to keep it optional, so perhaps `from` should be passed to `mabye_handle` too?

Reply:
It is optional because you may never call it. But if you call it, it has to be implemented. So the current code is correct!

Reply:
Ah, of course! 👍